### ISSUE TITLE
Get Yaw-angle from yolov8

### ISSUE THREAD

**Issue by LeonGitafk** [ultralytics/ultralytics#18442]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi,
i wonder how I can find out the yaw angle to the camera from a **2D image** of an airplane with Yolov8 or other Yolo models. 
I have my **own data set** of airplanes taken from different angles. 
I have already labeled them with normal bounding boxes. 
My idea was to change the .txt file for each image so that at the end the **angle of rotation (yaw)** that I had previously estimated visually on each image is added. 
Now the real question is how can I train the Yolo model correctly so that there are no problems and I can query the **yaw angle** of the aircraft to the camera at the end when applying the trained model. 
I think there would be problems with the format of the .txt file without a **special change** in the Yolo model if I simply add another value. 

This I think is the usual format: <class> <x1> <y1> <x2> <y2>
And this is what I thought about: <class> <x1> <y1> <x2> <y2> <yaw angle>

This is what I mean with the Yaw Angle and how I would describe it:
![explanation](https://github.com/user-attachments/assets/cb7f829c-9ff0-4d5d-84a4-7a225c665a4b)

### Additional

I work with Windows 10

**Comment by UltralyticsAssistant** [comment]
👋 Hello @LeonGitafk, thank you for your question and for your interest in using Ultralytics YOLO 🚀! 

If you’re new to Ultralytics, we recommend checking out the [Docs](https://docs.ultralytics.com), where you’ll find helpful guides for both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Many users find answers to common questions there.

### Custom Label Formats ❓

We understand that you are aiming to modify the YOLO annotation format to include an additional field for the yaw angle. It is important to note that modifying the standard YOLO annotation format (<class> <x1> <y1> <x2> <y2>) to include an extra field, such as the yaw angle, would require modifications to the underlying training and model architecture. This is not directly supported by default in the current version of YOLOv8. Could you provide further details or a minimum reproducible example (MRE) of your intended approach? For more on MREs, see our [help guide](https://docs.ultralytics.com/help/minimum_reproducible_example/) 😊.

If you are exploring custom implementations, you may need to dive into the [source code](https://github.com/ultralytics/ultralytics) and potentially adjust the Dataset loading, Model configuration, and Loss function to incorporate such changes. 

### Suggestions 💡

Until an Ultralytics engineer assists further, here's what you might consider:  
1. **Feature Augmentation**: Instead of modifying `.txt` files, consider appending the angle as metadata in your dataset that your model can leverage during training.  
2. **Custom Head**: If you need yaw angle predictions, explore adding an additional custom regression head to YOLOv8 for this task. Resources like [PyTorch Docs](https://pytorch.org/docs/stable/index.html) can help with building such custom layers.  
3. **Community Help**: You can also discuss your approach and get feedback from other users and developers in our [Ultralytics Discord](https://discord.com/invite/ultralytics) 🎧 or [Discourse forums](https://community.ultralytics.com).

### Getting Started 🚀

To ensure optimal performance, upgrade to the latest version of the `ultralytics` package. Run the following command in your **Python>=3.8** environment:

```bash
pip install -U ultralytics
```

YOLO models can be run in various supported environments with all necessary dependencies pre-installed. Explore these environments for your project:
- **Notebooks** offering free GPUs:  
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>  
- **Cloud**: Check out [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/) quickstart guides.  
- **Docker** images: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>  

### Continuous Integration ✅

Our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) system runs daily on macOS, Windows, and Ubuntu. Check the badge below to see if all tests currently pass:  
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

We hope this helps clarify your path forward! This is an automated response 🤖, but don’t worry—an Ultralytics engineer will assist you soon to help address your question more thoroughly. Feel free to share more details in the meantime to expedite support 🛠️.

**Comment by Y-T-G** [comment]
You would need to significantly change the source code from data loading to model architecture to loss functions.

**Comment by LeonGitafk** [comment]
Thank you for your answer! Is there possibly another way I could calculate the yaw angle? I was thinking of YOLO pose models, for example, where you work with keypoints. Could this achieve my goal if I re-label the data set?

**Comment by glenn-jocher** [comment]
@LeonGitafk using YOLO pose models with keypoints could indeed be a viable approach. By labeling the keypoints corresponding to the airplane's orientation (e.g., nose, tail, wings), you could infer the yaw angle based on their relative positions. This method would require re-labeling your dataset with the appropriate keypoints and training a YOLO pose model. You can learn more about pose estimation datasets and formats here: https://docs.ultralytics.com/datasets/pose/.

**Comment by LeonGitafk** [comment]
Thanks for your verification! I will start labelling the data now. Maybe I will ask for your help again when calculating the angle

**Comment by glenn-jocher** [comment]
You're welcome! Feel free to reach out if you have further questions while working on your dataset or calculating the angle. Best of luck with your labeling!
